### PR TITLE
update crashdump for upgrade and migration

### PR DESCRIPTION
**Pull request by grwilson**

### CODE REVIEW

**Comment by grwilson** [comment]
[ab-pre-push](http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/appliance-build-orchestrator-pre-push/1216/)

**Comment by prakashsurya** [comment]
@grwilson FYI, you can use `git-ab-pre-push` to test upgrade (and soon migration too) by using the `--test-upgrade-from` option; e.g. `--test-upgrade-from 5.3` or `--test-upgrade-from trunk`

File: live-build/misc/migration-scripts/dx_apply [hunk 1]
```diff
@@ -188,6 +188,7 @@ cat <<-EOF >"$TMP_ROOT/etc/fstab" || die "Failed to setup /etc/fstab"
 	$RPOOL/ROOT/$FSNAME/home /export/home zfs defaults,x-systemd.before=zfs-import-cache.service 0 0
 	$RPOOL/ROOT/$FSNAME/data /var/delphix zfs defaults,x-systemd.before=zfs-import-cache.service 0 0
 	$RPOOL/ROOT/$FSNAME/log  /var/log     zfs defaults,x-systemd.before=zfs-import-cache.service 0 0
+	rpool/crashdump          /var/crash   zfs defaults,x-systemd.before=zfs-import-cache.service,x-systemd.before=kdump-tools.service 0 0
```
Comment (pzakha) [inline live-build/misc/migration-scripts/dx_apply]: nit: keep naming consistent. We use `$RPOOL` elsewhere. IMO this indirection is not needed, but we should probably stay consistent.

**Comment by pzakha** [comment]
Can you also do migration testing once the image is built?

**Comment by grwilson** [comment]
[migration testing](http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/blackbox-chained/228/)
[ab-pre-push (with upgrade)](http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/appliance-build-orchestrator-pre-push/1218/)

**Pull request merged**